pieces/regex_json/README.md

```diff
@@ -210,3 +211,14 @@ test tests::test_run ... ok
 
 test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.01s
 ```
+
+## Takeaways
+
+In this eample the [anyhow](https://github.com/dtolnay/anyhow) crate is used. This crate simplifes
+error handling with the use of `anyhow::Result`, allowing the `?` operator to be used on all error
+types. Additionally the return signature is simplified because the error is implied. This means instead
```

Comment:
I think the term "implied" here doesn't do enough to explain what the `?` operator actually does. In my mind, the purpose of `?` operator under normal circumstances is to propagate errors back to the calling function, so that you need to only properly handle the error in the root caller.

Does `anyhow::Result` do it any different?

Reply:
I may have cut too much here. There are two things I’m trying to say. With the standard `Result` you can only use `?` if the error is the type specified in the return. So if the return is `Result<T, std::io::Error>` you can only use `?` if the resulting error would be an io error. With anyhow you can use `?` for any error. 

What I am trying to say with “implied” is instead returning `Result<T, std::io::Error>` you return `Result<T>` with no error specification. The error is understood to be an `anyhow::Error` which covers any error type.